plugins/providers/bigquery/client.go

```diff
@@ -110,38 +110,21 @@ func (c *bigQueryClient) GetTables(ctx context.Context, datasetID string) ([]*Ta
 	return results, nil
 }
 
-func (c *bigQueryClient) ResolveDatasetRole(role string) (bq.AccessRole, error) {
-	switch role {
-	case DatasetRoleReader:
-		return bq.ReaderRole, nil
-	case DatasetRoleWriter:
-		return bq.WriterRole, nil
-	case DatasetRoleOwner:
-		return bq.OwnerRole, nil
-	default:
-		return "", ErrInvalidRole
-	}
-}
-
 func (c *bigQueryClient) GrantDatasetAccess(ctx context.Context, d *Dataset, user, role string) error {
 	dataset := c.client.Dataset(d.DatasetID)
 	metadata, err := dataset.Metadata(ctx)
 	if err != nil {
 		return err
 	}
 
-	bqRole, err := c.ResolveDatasetRole(role)
-	if err != nil {
-		return err
-	}
 	for _, a := range metadata.Access {
-		if a.Entity == user && a.Role == bqRole {
+		if a.Entity == user && a.Role == bq.AccessRole(role) {
```

Comment:
is it possible if `a.Role="READER"` and `role="roles/bigquery.dataViewer"`? if so, it should still fulfill the condition checking here in a way because both are essentially the same role

Reply:
@rahmatrhd it'll not be the same, users should not use both to avoid ambiguity.

Reply:
handle the scenario